azurerm/data_factory.go

```diff
@@ -0,0 +1,74 @@
+package azurerm
+
+import (
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/datafactory/mgmt/2018-06-01/datafactory"
+)
+
+func expandDataFactoryParameters(input map[string]interface{}) map[string]*datafactory.ParameterSpecification {
+	output := make(map[string]*datafactory.ParameterSpecification)
+
+	for k, v := range input {
+		output[k] = &datafactory.ParameterSpecification{
+			Type:         datafactory.ParameterTypeString,
+			DefaultValue: v.(string),
+		}
+	}
+
+	return output
+}
+
+func expandDataFactoryAdditionalProperties(input map[string]interface{}) map[string]interface{} {
```

Comment:
Do we really need this? 🤔  should you be able to use the source map without copying it to a new one?

Reply:
There needs to be a facepalm emoji in github